### PR TITLE
Px4fix/master multi mavlink fix

### PR DESCRIPTION
**Pull request by oberion**
As discussed in Issue #2641, this is a fix of setting Waypoints over different mavlink links.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -4,7 +4,7 @@
 [submodule "NuttX"]
 	path = NuttX
 	url = git://github.com/PX4/NuttX.git
-[submodule "libuavcan"]
+[submodule "src/lib/uavcan"]
```
Comment (LorenzMeier) [inline .gitmodules]: Why is this in here?

File: src/modules/mavlink/mavlink_mission.cpp [hunk 2]
```diff
@@ -70,14 +76,12 @@ MavlinkMissionManager::MavlinkMissionManager(Mavlink *mavlink) : MavlinkStream(m
...
-	_dataman_id(0),
-	_count(0),
-	_current_seq(0),
+	_my_dataman_id(0),
 	_transfer_dataman_id(0),
 	_transfer_count(0),
 	_transfer_seq(0),
 	_transfer_current_seq(0),
-	_transfer_partner_sysid(0),
+	_transfer_partner_sysid(255),
```
Comment (LorenzMeier) [inline src/modules/mavlink/mavlink_mission.cpp]: That's a hack - it should default to 0 and should work that way, because its set on transfer init.

File: .gitmodules (hunk 1 above)
Comment (oberion) [inline .gitmodules]: OK, this is not related to the fix. I has to change this in order to successfully update the submodule. Without the changes Tortoisegit was not able to update this submodule on my windows machine.

Comment (LorenzMeier) [inline .gitmodules]: never mind, then we'll bring that in as a drive-by. I just need to validate it.
